File: INIGO   SOLOMON/desk/Thrift try/AdditionServer.java
import org.apache.thrift.server.TServer;
import org.apache.thrift.server.TServer.Args;
import org.apache.thrift.server.TSimpleServer;
import org.apache.thrift.transport.TServerSocket;
import org.apache.thrift.transport.TServerTransport;

public class AdditionServer {

  public static AdditionHandler handler;
  public static AdditionService.Processor processor;
  public static void main(String[] args){
    try {
        handler=new AdditionHandler();
        processor=new AdditionService.Process(handler);
         
	Runnable simple=new Runnable() {
	  public void run() {
	     simple(processor);
	   }
	  };
	   
	    new Thread(simple).start();
	  }catch(Exception x) {
	    x.printStackTrace();
	  }
	}
	
	public static void simple(AdditionService.Processor processor) {
	  try {
	 
	    TServerTransport serverTransport=new TServerSocket(9090);
	    TServer server=new TSimpleServer(new Args(serverTransport).processor(processor));

	     System.out.println("starting the simple server");
	     server.serve();
	   }catch(Exception e) {
	    e.printStackTrace();
	   }
	}
       }

File: INIGO   SOLOMON/desk/py-impl/AdditionServer.py
import sys
sys.path.append('/home/sanjyot/Desktop/gen-py') 
from add import AdditionService
from add.ttypes import *
 
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol
from thrift.server import TServer

import socket
 
class AdditionServiceHandler:
  def __init__(self):
    self.log = {}
 
  def ping(self):
    print "ping()"
 

 
  def sayMsg(self, msg):
    
    return "say " + msg + " from " + socket.gethostbyname(socket.gethostname())
 
handler = AdditionServiceHandler()
processor = AdditionService.Processor(handler)
transport = TSocket.TServerSocket(9090)
tfactory = TTransport.TBufferedTransportFactory()
pfactory = TBinaryProtocol.TBinaryProtocolFactory()
 
server = TServer.TSimpleServer(processor, transport, tfactory, pfactory)
 
print "Starting python server..."
server.serve()
print "done!"
